lib/spack/spack/build_systems/oneapi.py

```diff
@@ -58,7 +58,8 @@ def component_dir(self):
     @property
     def component_prefix(self):
         """Path to component <prefix>/<component>/<version>."""
-        return self.prefix.join(join_path(self.component_dir, self.spec.version))
+        v = self.spec.version if self.spec.satisfies("@:2023") else self.spec.version.up_to(2)
```

Comment:
This works for the compiler and mkl, but not mpi and tbb which 2021.x and 2022.x. I have added a property to each package with the version number of the new layout. I will replace my hacky split/join and use your up_to(2) to make the new directory.